src/Gui/DockWindowManager.cpp

```diff
@@ -380,6 +391,24 @@ void DockWindowManager::setup(DockWindowItems* items)
         }
     }
 
+    // Don't always tabify after switching the workbench
+    static bool tabify = false;
+    if (!tabify) {

```

Comment:
@wwmayer I'm not clear about that.
The comment says `// Dont' always` but from the code the logic makes that it can happen only once : `tabify` can never become `false` again after it has been `true`. Can you tell what is the functional goal of this ?

Also if I'm correct, the rest of the function above does nothing real. So shouldn't we also prevent it to run if `tabify` is `true` ?
Thx

Reply:
It is supposed to happen only once. When a user has modified the default layout he for sure don't want to restore it every time he switches to another workbench, no?

> Also if I'm correct,

Correct.

Reply:
> It is supposed to happen only once. When a user has modified the default layout he for sure don't want to restore it every time he switches to another workbench, no?

Sure. I would have been no doubt about implementation if the comment said `only once`. :wink: The `Not always` make we wonder about the implementation correctness. ;)
One last point fo rmy full understanding. I guess that if "tabification" shall happen, it should do only at first call (when WB layout is initialized).
But with current implementation, `tabify` could stay `false` if WB initializes with no tabbed widgets. Well, OK, in this case `tabifyDockWidgets` should always do nothing, but it will be completely processed each time. Couldn't we make `tabify = true` unconditionally at the end of the function so it will return immediately after 1st call ?
Thx -- sorry for this :smile: --

Reply:
>But with current implementation, tabify could stay false if WB initializes with no tabbed widgets.

Yes. During the startup phase of FreeCAD an empty workbench is used for the initialization because at this point the real workbench cannot be used yet. That's why the `tabify` variable is only changed if something has happened.

In the worst case what could happen (none of the workbenches have dock widgets that can be tabbed) the function will be called each time after switching a workbench and it will never do anything. But is this a problem? I don't think so.

> Couldn't we make tabify = true unconditionally at the end of the function so it will return immediately after 1st call ?

Sorry but I don't get your point?

Reply:
> Yes. During the startup phase of FreeCAD an empty workbench is used for the initialization because at this point the real workbench cannot be used yet. That's why the `tabify` variable is only changed if something has happened.

Thanks, was precisely the information I was missing.

> In the worst case what could happen (none of the workbenches have dock widgets that can be tabbed) the function will be called each time after switching a workbench and it will never do anything. But is this a problem? I don't think so.

Not a problem per se. :) Just seeking for code meaningfulness to help future maintenance. With your above explanation, good as is. :+1:

> > Couldn't we make tabify = true unconditionally at the end of the function so it will return immediately after 1st call ?
> 
> Sorry but I don't get your point?

I thought it shall be call only once + at first call. With your above explanation doesn't make much sense anymore.

Thanks for the details. Going to merge.